src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2422,51 +2422,81 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		}
 		bindHandler(Opcode.RESUME); {
 			var resumeStub = X86_64Label.new();
-			var cont_id = r_tmp0, r_ip_store = r_tmp1;
+			var cont_id = xenv.tmp0, ip_store = xenv.tmp1;
 			computeCurIpForTrap(-1);
 			computePcFromCurIp();
 			// read info and rewind %ip (need %ip at start of opcode during call)
-			asm.movq_r_r(r_ip_store, r_ip);
-			genReadUleb32(cont_id); // read signature index
-			asm.movq_r_r(r_ip, r_ip_store);
+			masm.emit_mov_r_r(ValueKind.REF, ip_store, xenv.ip);
```

Comment:
I don't quite understand why you need to store the `ip` from before the read of the continuation index.

Reply:
I think this is because when handling a suspension `findSuspensionHandler` uses the parent stack's `ip` (which is excepted to point to the beginning of `resume`) to find the corresponding handler. In `V3Interpreter` this is done with a field that records the previous `resume`'s instruction pointer.

Reply:
This lookup should be based on the `pc` stored in the frame, not on the direct instruction pointer, which is a pointer into the middle of the bytecode.